std/random.d

```diff
@@ -1336,6 +1330,297 @@ version (AnyARC4Random)
+    private alias DWORD = size_t; // uint in druntime
+
+    private alias FnCryptAcquireContextW = extern(Windows)
+        BOOL function(HCRYPTPROV*, LPCWSTR, LPCWSTR, DWORD, DWORD) @nogc nothrow;
+    private alias FnCryptAcquireContextA = extern (Windows)
+        BOOL function(HCRYPTPROV*, LPCSTR, LPCSTR, DWORD, DWORD) @nogc nothrow;
+    private alias FnCryptGenRandom = extern(Windows)
+        BOOL function(HCRYPTPROV, DWORD, scope PBYTE) @nogc nothrow;
+    private alias FnReleaseContext = extern(Windows)
+        BOOL function(HCRYPTPROV, DWORD) @nogc nothrow;
+
+    /+
+    Tries to use a system API to get entropy. Resources are not cached
+    due to the expectation that this function will not be called more
+    than once per program execution.
+
+    Params:
+        ptr = address at which to start writing
+        len = number of bytes to write
+
+    Returns:
+    `len` on success, -1 on failure
+    +/
+    private ptrdiff_t oneTimeGenRandom()(scope void* ptr, ptrdiff_t len) @system
+    {
+        import core.sys.windows.winbase : GetLastError;
+        import core.sys.windows.winerror : NTE_BAD_KEYSET;
+        import core.sys.windows.wincrypt : PROV_RSA_FULL, CRYPT_NEWKEYSET,
+            CRYPT_VERIFYCONTEXT, CRYPT_SILENT;
+        auto hAdvapi32 = LoadLibraryA("Advapi32.dll");
+        if (!hAdvapi32) return -1;
+        scope (exit) { if (hAdvapi32) FreeLibrary(hAdvapi32); }
+        ULONG_PTR hProvider;
+        auto fnCryptReleaseContext = cast(FnReleaseContext)
+            GetProcAddress(hAdvapi32, "CryptReleaseContext");
+        if (!fnCryptReleaseContext) return -1;
+        auto fnCryptAcquireContextW = cast(FnCryptAcquireContextW)
+            GetProcAddress(hAdvapi32, "CryptAcquireContextW");
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379886(v=vs.85).aspx
+        // For performance reasons, we recommend that you set the pszContainer
+        // parameter to NULL and the dwFlags parameter to CRYPT_VERIFYCONTEXT
+        // in all situations where you do not require a persisted key.
+        // CRYPT_SILENT is intended for use with applications for which the UI
+        // cannot be displayed by the CSP.
+        if (!(fnCryptAcquireContextW && fnCryptAcquireContextW(
+                &hProvider, null, null,
+                PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT)))
+        {
+            if (GetLastError() == NTE_BAD_KEYSET)
+            {
+                // Attempt to create default container
+                auto fnCryptAcquireContextA = cast(FnCryptAcquireContextW)
+                    GetProcAddress(hAdvapi32, "CryptAcquireContextA");
+                if (!(fnCryptAcquireContextA && fnCryptAcquireContextA(
+                    &hProvider, null, null,
+                    PROV_RSA_FULL, CRYPT_NEWKEYSET | CRYPT_SILENT)))
+                {
+                    return -1;
+                }
+            }
+        }
+        assert(hProvider);
```

Comment:
For Phobos, we try to add messages to `assert` so that in case they see it (e.g. and don't have line number on their stack trace working like on OSX), they have a good start where too look. Also sometimes the assert already tells enough, s.t. Phobos's code doesn't need to be looked at.

Reply:
fixed